p2p/pex/pex_reactor.go

```diff
@@ -338,9 +341,22 @@ func (r *PEXReactor) ensurePeers() {
 		toDial[try.ID] = try
 	}
 
+	// 1s == (1e9 ns) == (1 Billion ns)
+	billionNs := float64(time.Second.Nanoseconds())
+
 	// Dial picked addresses
-	for _, item := range toDial {
+	for _, addr := range toDial {
 		go func(picked *p2p.NetAddress) {
+			// exponential backoff if it's not our first attempt to dial picked address
+			var attempt int
+			if lAttempt, notFirstAttempt := r.attemptsToDial.Load(picked.DialString()); notFirstAttempt {
```

Comment:
Some minor naming nits. Would call `attempt`, `attempts` here as we store the amount of times we attempted to connect. And `notFirstAttempt` -> `attempted` for more concise and inverse of the meaning.